### PR TITLE
8261131: jcmd jmap dump should not accept gz option with no value

### PR DESCRIPTION
**Pull request by linzang**
---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8261131](https://bugs.openjdk.java.net/browse/JDK-8261131): jcmd jmap dump should not accept gz option with no value

### Reviewers
 * [Chris Plummer](https://openjdk.java.net/census#cjplummer) (@plummercj - **Reviewer**) ⚠️ Review applies to 620423efc07b5dd3f070140056288684f5c4a125
 * [Serguei Spitsyn](https://openjdk.java.net/census#sspitsyn) (@sspitsyn - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/2399/head:pull/2399`
`$ git checkout pull/2399`

### CODE REVIEW

File: src/jdk.jcmd/share/classes/sun/tools/jmap/JMap.java [hunk 1]
```diff
@@ -227,7 +227,7 @@ private static void dump(String pid, String options)
                 // ignore format (not needed at this time)
             } else if (subopt.startsWith("gz=")) {
                compress_level = subopt.substring("gz=".length());
-               if (compress_level == null) {
+               if (compress_level == null || compress_level.length() == 0) {
```
Comment (plummercj) [inline src/jdk.jcmd/share/classes/sun/tools/jmap/JMap.java:230]: I don't think `substring` can return `null`, so the proper fix was to remove the null check and replace with your check for `== 0`.

**Review by plummercj — COMMENTED** [review]
Copyrights needs updating in both files.

**Review by plummercj — APPROVED** [review]

**Comment by linzang** [comment]
Thanks @plummercj for help reviewing this PR! mark it as ready for push.

**Review by sspitsyn — APPROVED** [review]
Hi Lin,
The fix looks good to me.
Thanks,
Serguei

**Comment by linzang** [comment]
Dear All，
May I ask your help to push this PR as it is ready. 
Thanks!
Lin

**Pull request closed without merging**